diffsync/utils.py

```diff
@@ -42,3 +47,44 @@ def __missing__(self, key):
         """When trying to access a nonexistent key, initialize the key value based on the internal factory."""
         self[key] = value = self.factory()
         return value
+
+
+# from: https://stackoverflow.com/questions/72618673/list-directory-tree-structure-in-python-from-a-list-of-path-file
+def _tree(data: dict, prefix: str = ""):
```

Comment:
We need to use `typing.Dict` rather than `builtins.dict` as long as we have to support a Python version <3.9